### PR TITLE
feat: add get session id

### LINKED ISSUE #133: Expose "sessionId" to Flutter Package API
### Problem Statement

Hello, I need to access the user's current session ID to link to some logs. From what I've seen, this function is already implemented in both the [Android](https://github.com/PostHog/posthog-android/blob/main/posthog/src/main/java/com/posthog/internal/PostHogSessionManager.kt#L33-L39) and [iOS](https://github.com/PostHog/posthog-ios/blob/main/PostHog/PostHogSessionManager.swift#L23-L29) SDKs. It would be an idea to add this to the Flutter SDK. I can help with the implementation if is needed.

### Solution Brainstorm

_No response_

### PR DESCRIPTION
**Pull request by GabrielRStabile**
## :bulb: Motivation and Context
Resolve #133 

## :pencil: Checklist

- [X] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [X] No breaking change or entry added to the changelog.

### CODE REVIEW

File: android/src/main/kotlin/com/posthog/flutter/PosthogFlutterPlugin.kt [hunk 1]
```diff
@@ -500,6 +503,15 @@ class PosthogFlutterPlugin :
         }
     }
 
+    private fun getSessionId(result: Result) {
+        try {
+            val sessionId = PostHog.getSessionId()
+            result.success(sessionId.toString())
+        } catch (e: Exception) {
+            result.error("ERROR", "Error getting session id", e.message)
```
Comment (marandaneto) [inline android/src/main/kotlin/com/posthog/flutter/PosthogFlutterPlugin.kt:511]:
```suggestion
            result.error("PosthogFlutterException", e.localizedMessage, null)
```

File: android/src/main/kotlin/com/posthog/flutter/PosthogFlutterPlugin.kt [hunk 2]
```diff
@@ -500,6 +503,15 @@ class PosthogFlutterPlugin :
         }
     }
 
+    private fun getSessionId(result: Result) {
+        try {
+            val sessionId = PostHog.getSessionId()
+            result.success(sessionId.toString())
```
Comment (marandaneto) [inline android/src/main/kotlin/com/posthog/flutter/PosthogFlutterPlugin.kt:509]:
```suggestion
            result.success(sessionId?.toString())
```

sessionId can be null here

File: lib/src/posthog_flutter_web_handler.dart [hunk 3]
```diff
@@ -94,6 +94,9 @@ Future<dynamic> handleWebMethodCall(MethodCall call, JsObject context) async {
         call.arguments['key'],
       ]);
       break;
+    case 'getSessionId':
+      final sessionId = analytics.callMethod('get_session_id');
+      return sessionId;
```
Comment (marandaneto) [inline lib/src/posthog_flutter_web_handler.dart:99]: https://github.com/PostHog/posthog-js/blob/6da785b18c526f5008c2ec9f0bc98555ca784404/src/posthog-core.ts#L1616
sessionId can be empty, lets return null if its empty instead

**Comment by marandaneto** [comment]
Thanks @GabrielRStabile 
Mind addign an entry to https://github.com/PostHog/posthog-flutter/blob/main/CHANGELOG.md ?
I will test, merge, and release it by the EoW.

**Review by marandaneto — APPROVED** [review]

**Comment by GabrielRStabile** [comment]
> Thanks @GabrielRStabile Mind adding an entry to https://github.com/PostHog/posthog-flutter/blob/main/CHANGELOG.md ? I will test, merge, and release it by the EoW.

Done!

File: android/src/main/kotlin/com/posthog/flutter/PosthogFlutterPlugin.kt [hunk 4]
```diff
@@ -500,6 +503,15 @@ class PosthogFlutterPlugin :
         }
     }
 
+    private fun getSessionId(result: Result) {
+        try {
+            val sessionId = PostHog.getSessionId()
+            result.success(sessionId?.toString())
+        } catch (e: Exception) {
```
Comment (marandaneto) [inline android/src/main/kotlin/com/posthog/flutter/PosthogFlutterPlugin.kt:510]:
```suggestion
        } catch (e: Throwable) {
```

**Comment by marandaneto** [comment]
> @marandaneto can you please merge this if it's all done

I pushed 2 commits to your branch since there were missing things

**Review by marandaneto — APPROVED** [review]
Thanks @GabrielRStabile

**Pull request merged**